linker/src/main/kotlin/com/openlattice/linking/controllers/RealtimeLinkingController.kt

```diff
@@ -77,4 +88,59 @@ class RealtimeLinkingController(
 
         return matchedEntityPairs
     }
+
+    @RequestMapping(
+            path = [RealtimeLinkingApi.BLOCKING],
+            method = [RequestMethod.POST],
+            produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun block(
+            @RequestBody blockingRequest: BlockingRequest
+    ): Map<UUID, Map<UUID, List<BlockedEntity>>> {
+        ensureAdminAccess()
+        val propertyFqnCache = CacheBuilder
+                .newBuilder()
+                .maximumSize(8192)
+                .build(CacheLoader.from { propertyTypeId: UUID? -> edmService.getPropertyTypeFqn(propertyTypeId!!) })
+
+        return blockingRequest.entities.mapValues { (entitySetId, entityKeyIds) ->
+            entityKeyIds.associateWith { entityKeyId ->
+                blocker
+                        .block(EntityDataKey(entitySetId, entityKeyId),top = blockingRequest.blockSize)
```

Comment:
what is `,top`?

Reply:
the size of the blocks.

Reply:
i.e it controls how many elasticsearch results are requested

Reply:
cool, makes sense!